### PR TITLE
[FR0-9] Rating Component

### PR DESCRIPTION
**Pull request by Rmichal503**
Created Rating Component with two variants: :heart: and :star:. 
Each variant have three types: default, number and hidden (on screenshot from top: hidden, default, number).
Icons are generated based on "rate" prop
Zrzut ekranu 2022-12-1 o 12 43 27
PR is without test file because Jest doesn't like svgr (for now 😛)

### CODE REVIEW

File: src/components/Rating/Rating.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            type:{
+                name:'number',
+                require:true
+            },
+            
+        }
+    }
+  } as Meta;
+
+const Template: Story = (args) => <Rating rate={0} {...args}/>
```
Comment (patryk-bernasiewicz) [inline src/components/Rating/Rating.stories.tsx:39]: `rate` prop comes from `args` so it doesn't have to explicitly set here

File: src/components/Rating/Rating.test.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+describe('Rating Component',()=>{
+    it('Renders image', ()=>{
+        render(<Rating rate={2} />);
+        expect(screen.findByRole('heading'))
+    })
+    // it('Not renders image', ()=>{
+    //     const numOfRate:number = 0
+    //     render(<Rating rate={numOfRate}/>)
+    //     expect(screen.getAllByRole('img')).not.toBeInTheDocument()
+    // })
```
Comment (patryk-bernasiewicz) [inline src/components/Rating/Rating.test.tsx:14]: This test can be removed completely as we're not going to use `img`s

File: src/components/Rating/Rating.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import Stars from '@svg/star.svg'
+
+export type RatingProps = { variant?: string; type?: string; rate: number };
+
+export const Rating = (props: RatingProps) => {
+  const { variant , type = 'default', rate } = props;
+
+  return (
+    <>
+    {rate>0?<div key={`${type}${variant}${rate}`} className="flex items-center">
```
Comment (patryk-bernasiewicz) [inline src/components/Rating/Rating.tsx:11]: You can simplify and flatten this condition by moving it above the default render, eg.

```typescript
if (rate <= 0) {
    return null;
}

return (
    <div>...</div>
);

```

File: src/components/Rating/Rating.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import Heart from '@svg/heart.svg'
+import Stars from '@svg/star.svg'
+
+export type RatingProps = { variant?: string; type?: string; rate: number };
```
Comment (patryk-bernasiewicz) [inline src/components/Rating/Rating.tsx:4]: `variant` should explicitly have type of either 'stars' or 'hearts'. Same goes for `type` - it should list only the allowed strings.

File: src/components/Rating/Rating.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+export const Rating = (props: RatingProps) => {
+  const { variant , type = 'default', rate } = props;
```
Comment (patryk-bernasiewicz) [inline src/components/Rating/Rating.tsx:7]: You can perform destructurization directly in the parentheses, but that's a matter of preference :)

File: src/components/Rating/Rating.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        (el, index) => {
+          return (
+            <>
+            {variant==='hearts'?<Heart key={`rate${variant}${Math.random()}`}
+              className={
+                index + 1 <= rate
+                  ? 'mx-px'
+                  : type === 'hidden'
+                  ? 'hidden'
+                  : 'mx-px opacity-50'
```
Comment (patryk-bernasiewicz) [inline src/components/Rating/Rating.tsx:22]: You could store such condition in a variable, eg.

```typescript
const isActive = index < rate;
const isHidden = type === 'hidden';
```
and then use [clsx](https://www.npmjs.com/package/clsx) package to join classes with conditions
```typescript
import clsx from 'clsx';

const iconClasses = clsx(
    'mx-px',
    isHidden && 'hidden',
    !isActive && !isHidden && 'opacity-50'
);
```

The way it works is it joins the strings which have a truthy value on the left side of their condition. See more in the package's README: [https://github.com/lukeed/clsx#readme](github: clsx)

File: src/components/Rating/Rating.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+              }/>}
+            </>
+          );
+        },
+      )}
+      {type === 'number' ? <span>{rate}</span> : null}
```
Comment (patryk-bernasiewicz) [inline src/components/Rating/Rating.tsx:35]: You can use simplified conditional rendering like 

```tsx
{type === 'number' && <span>{rate}</span>}
```

**Review by patryk-bernasiewicz — CHANGES_REQUESTED** [review]
The PR got messed up by my changes in the `main` branch, but that's okay - the next ones should be working properly

File: src/components/Rating/Rating.stories.tsx (hunk 1 above)
Comment (Rmichal503) [reply src/components/Rating/Rating.stories.tsx:39]: without `rate` prop I've got error. But I checked now without this prop and it's working perfect, with error in vsc but works :D

**Review by patryk-bernasiewicz — APPROVED** [review]
Looks great! 🚀

**Review by bialek-k — APPROVED** [review]

**Pull request merged**